server/data/cache/SmokeTestStore.ts

```diff
@@ -0,0 +1,35 @@
+import { Request } from 'express'
+import redis from 'redis'
+import crypto from 'crypto'
+import createRedisClient from './createRedisClient'
+import RedisStore from './RedisStore'
+import config from '../../config'
+
+const SMOKE_TEST = 'smokeTest'
+
+export default class SmokeTestStore extends RedisStore {
+  constructor(redisClient: redis.RedisClient = createRedisClient(`${SMOKE_TEST}:`)) {
+    super(redisClient)
+  }
+
+  async setSmokeTestSecret(oneTimeSecret: string): Promise<void> {
+    return this.setRedisAsync(SMOKE_TEST, oneTimeSecret, 'EX', 60)
+  }
+
+  public async getSmokeTestSecret(): Promise<string> {
+    const secret = this.getRedisAsync(SMOKE_TEST)
+    this.deleteEntry(SMOKE_TEST)
+    return secret
+  }
+
+  public async startSmokeTest(req: Request): Promise<string> {
```

Comment:
Minor, and we can save it for another day, but I'm not sure about this redis store having knowledge of express and its `Request` type.
I wonder if the method signature could just take the msjSecret as a string, and the thing thats calling this be responsible for getting it out of the request body?
One for another day ... 👍